server/internal/openapi/process.go

```diff
@@ -138,6 +138,30 @@ func (p *ToolExtractor) Do(
 		return nil, oops.E(oops.CodeInvariantViolation, oops.Permanent(err), "unable to process openapi document").Log(ctx, p.logger)
 	}
 
+	var (
+		doc     []byte
+		readErr error
+	)
+	func() {
```

Comment:
Let's not nest anonymous functions that close over variables in Go. Extract to a separate function.